src/Core/Services/VaultTimeoutService.cs

```diff
@@ -214,15 +223,30 @@ public async Task SetVaultTimeoutOptionsAsync(int? timeout, VaultTimeoutAction?
         {
             await _stateService.SetVaultTimeoutAsync(timeout);
             await _stateService.SetVaultTimeoutActionAsync(action);
-            await _cryptoService.ToggleKeyAsync();
+            await _cryptoService.ToggleKeysAsync();
             await _tokenService.ToggleTokensAsync();
         }
 
-        public async Task<Tuple<bool, bool>> IsPinLockSetAsync(string userId = null)
+        public async Task<PinLockEnum> IsPinLockSetAsync(string userId = null)
         {
-            var protectedPin = await _stateService.GetProtectedPinAsync(userId);
-            var pinProtectedKey = await _stateService.GetPinProtectedAsync(userId);
-            return new Tuple<bool, bool>(protectedPin != null, pinProtectedKey != null);
+            // we can't depend on only the protected pin being set because old
+            // versions only used it for MP on Restart
+            var pinIsEnabled = await _stateService.GetProtectedPinAsync(userId);
+            var userKeyPin = await _stateService.GetUserKeyPinAsync(userId);
+            var oldUserKeyPin = await _stateService.GetPinProtectedAsync(userId);
```

Comment:
⛏️ I'd adjust the naming, when things start with `Is...` IMO it's supposed to return a `bool` value, so now it should be transformed to `GetPinLockStateAsync(...)` or something alike (depending on how you rename `PinLockEnum`).
The same applies inside, e.g. `GetProtectedPinAsync(...)` doesn't return a `bool` value so the variable name should be `protectedPin`, because as a matter of fact as it's in the code now `pinIsEnabled` doesn't tell you right away if the pin is enabled or not.